### DISCUSSION TITLE
[Feature] Knowledge Retrival for multiple inputs

### DISCUSSION THREAD

**Discussion by bing0037** [langgenius/dify#4419] [Suggestion]
### Self Checks

- [X] I have searched for existing issues [search for existing issues](https://github.com/langgenius/dify/issues), including closed ones.
- [X] I confirm that I am using English to submit this report (我已阅读并同意 [Language Policy](https://github.com/langgenius/dify/issues/1542)).
- [X] Please do not modify this template :) and fill in all the required fields.

### 1. Is this request related to a challenge you're experiencing?

I am trying to use dify to provide working steps by using the provided tools. To be specific, I use LLM to provide instructions to complete a mission; then I use knowledge retrival to align corresponding tools to meet each instruction. However, I don't know how many instructions could be provided at the workflow design stage.
我尝试使用Dify来实现操作步骤的执行和工具的自动选取。首先，对于一个任务，我用LLM获得了实现的步骤（假设为N步）；然后，对于每一步，我希望能够通过知识库检索（知识库中包含了对工具功能的描述），选取出应该使用的工具；然后，通过填入执行参数，实现每一步的运行以及整体任务的完成。

### 2. Describe the feature you'd like to see

For the knowledge retrival, I wish the input could be List type and each element of the input could be matched to a knowledge according to the knowledge base. Also, the output of the knowledge retrival node is also List type and contains matching results of each input request. This helps when the number of the knowledge retrival requests is unknown before execution. 
对于知识库检索节点，输入的类型支持List格式，并且能够对于List的每一个元素，进行检索，并将检索的结果填入Result的对应元素中。这样，能够实现在检索次数未知的情况下，自动完成检索。

### 3. How will this feature improve your workflow or experience?

This could make tool selection and knowledge retrival available and the whole project feasible. Without this functionality, I may have to code by myself instead of using workflow GUI.
如果没有这个功能，那就只能通过代码的形式，通过for循环实现知识检索了，那显然Dify就不能用了。

### 5. Can you help us with this feature?

- [X] I am interested in contributing to this feature.

**Comment by BradKML** [comment]
Is similar to mixing two different RAG data collections?
